jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/core/engines/AggregatingTestPlanStats.java

```diff
@@ -43,6 +43,10 @@ private void addResult(SampleResult result) {
         stats.incErrors();
       }
       updateElapsedTime(result.getTime());
+      updateLatencyTime(result.getLatency());
+      updateProcessingTime(result.getTime() - result.getLatency());
```

Comment:
This is not necessarily processingTime, it might also include network or even server processing time. 

You could see elapsed time as: connection time + requestProcessingTime (library generating bytes to send and start sending through network + time to put all bytes in network) + request network time + server request processing time (receives all bytes, parses them, does actual processing, generates response) + server start sending response to network + time server takes to put all response bytes to network + network sending response time + response parsing/processing time.

Latency takes all that in consideration except the time between first byte is received and response parsing ends. Which might include server, network or JMeter (and used libraries) introduced delays.

I don't know a proper name for this metric. But, do you think is actually needed considering that you are already adding latency and you already have elapsed?

Reply:
I know f you remove it you might be "losing" data, since from aggregated results of other metrics you can't infer the max "processing" time.

Reply:
My main point is that processing might be interpreted as time spent in pre & post processors, which is not the case. 

I additionally wonder if by adding this we are not collecting too much info for most cases (percentiles are a little expensive), and maybe confusing beginner users. Maybe makes sense to use some default metrics, and enable some feature that enables this more detailed reporting or provide some extensibility option so users can implement their own stats collector? Not sure yet.

Additionally, if we decide to add these new methods, we need to add them to BZ module as well for it to compile.

Reply:
Aprox calculated increase in memory due to additional percentiles: from 480KB to 1.44MB. Is that much.